File: 백준/Silver/10815. 숫자 카드/숫자 카드.cc
#include <bits/stdc++.h>
using namespace std;

int main(void)
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;
    vector<int> arr;
    for(int i=0; i<n; i++) {
        int data;
        cin >> data;
        arr.push_back(data);
    }

    sort(arr.begin(), arr.end());

    int m;
    cin >> m;
    while(m--) {
        int target;
        cin >> target;
        cout << binary_search(arr.begin(), arr.end(), target) << " ";
    }

    return 0;
}

File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    static int n, m;
    static int[] arr;

    public static int binarySearch(int target) {
        int st = 0;
        int en = n-1;

        while(st <= en) {
            int mid = (st + en) / 2;

            if(arr[mid] == target) {
                return 1;
            }

            else if(arr[mid] > target) {
                en = mid-1;
            }

            else {
                st = mid+1;
            }
        }

        return  0;
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        n = scanner.nextInt();
        arr = new int[n];
        for(int i=0; i<n; i++) {
            arr[i] = scanner.nextInt();
        }
        Arrays.sort(arr);

        m = scanner.nextInt();
        for(int i=0; i<m; i++) {
            int target = scanner.nextInt();
            System.out.print(binarySearch(target) + " ");
        }
    }
}
